pkg/cli/admin/nodeimage/monitor.go

```diff
@@ -1,52 +1,368 @@
 package nodeimage
 
 import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"k8s.io/klog/v2"
+
 	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubectl/pkg/cmd/exec"
+	"k8s.io/kubectl/pkg/cmd/logs"
 	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/polymorphichelpers"
 	"k8s.io/kubectl/pkg/util/templates"
+	utilsexec "k8s.io/utils/exec"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+)
+
+const (
+	nodeJoinerMonitorContainer = "node-joiner-monitor"
 )
 
+var (
+	monitorLong = templates.LongDesc(`
+		Monitor nodes being added to a cluster using an image generated from
+		the "oc adm node-image create" command.
+
+		Each node being added to the cluster has two certificate signing requests
```

Comment:
Before citing this paragraph, I think that it could be useful to describe the fact that the monitor command will allow the user to observe the whole workflow. In particular, during the initial phase, when the pre-flight validations are being applied (and in case, if something went wrong, reported to the user)